src/nvim/undo.c

```diff
@@ -928,9 +939,129 @@ static u_header_T *unserialize_uhp(bufinfo_T *bi,
     return NULL;
   }
 
+  // Unserialize all extmark undo information
+  ExtmarkUndoObject *extup;
+  extmark_undo_vec_t uh_extmark;
+  kv_init(uh_extmark);
+
+  while ((c = undo_read_2c(bi)) == UF_ENTRY_MAGIC) {
+    bool error = false;
+    extup = unserialize_extmark(bi, &error, file_name);
+    if (error) {
+      kv_destroy(uh_extmark);
+      return NULL;
+    }
+    kv_push(uh_extmark, *extup);
+  }
+
+  uhp->uh_extmark = uh_extmark;
+
   return uhp;
 }
 
+static bool serialize_extmark(bufinfo_T *bi, ExtmarkUndoObject extup)
+{
+  ILOG("-------------------------");
+
+  undo_write_bytes(bi, (uintmax_t)extup.type, 4);
+  ILOG("%d", (int)extup.type);
+
+  if (extup.type == kExtmarkSplice) {
+      ILOG("Splicing:");
+    if (!undo_write(bi, (uint8_t *)&(extup.data.splice), sizeof(ExtmarkSplice))) {
+        return false;
+    }
+  } else if (extup.type == kExtmarkMove) {
+      ILOG("Moving:");
+    if (!undo_write(bi, (uint8_t *)&(extup.data.move), sizeof(ExtmarkMove))) {
+      return false;
+    }
+  } else if (extup.type == kExtmarkUpdate) {
+      ILOG("Updating:");
+    if (!undo_write(bi, (uint8_t *)&(extup.data.savepos), sizeof(ExtmarkSavePos))) {
+      return false;
+    }
+  } else if (extup.type == kExtmarkSavePos) {
+      ILOG("Saving Position:");
+    if (!undo_write(bi, (uint8_t *)&(extup.data.savepos), sizeof(ExtmarkSavePos))) {
+      return false;
+    }
+  }
```

Comment:
Not sure which union element corresponds to `kExtmarkClear` and `kExtmarkUpdate`.

Reply:
those are dead definitions, you can delete them.